UnitTests/Onboarding/ContextualOnboarding/BrowserTabViewControllerOnboardingTests.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  BrowserTabViewControllerOnboardingTests.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+import SwiftUI
+import Onboarding
+import Combine
+import BrowserServicesKit
+@testable import DuckDuckGo_Privacy_Browser
+
+final class BrowserTabViewControllerOnboardingTests: XCTestCase {
+
+    var viewController: BrowserTabViewController!
+    var dialogProvider: MockDialogsProvider!
+    var factory: CapturingDialogFactory!
+    var tab: Tab!
+    var cancellables: Set<AnyCancellable> = []
+    let expectation = XCTestExpectation()
+
+    @MainActor override func setUpWithError() throws {
+        try super.setUpWithError()
+        let tabCollectionViewModel = TabCollectionViewModel()
+        dialogProvider = MockDialogsProvider()
+        factory = CapturingDialogFactory(expectation: expectation)
+        tab = Tab()
+        tab.setContent(.url(URL.duckDuckGo, credential: nil, source: .appOpenUrl))
+        let tabViewModel = TabViewModel(tab: tab)
+        viewController = BrowserTabViewController(tabCollectionViewModel: tabCollectionViewModel, onboardingDialogTypeProvider: dialogProvider, onboardingDialogFactory: factory, featureFlagger: MockFeatureFlagger())
+        viewController.tabViewModel = tabViewModel
+        let window = NSWindow()
+        window.contentViewController = viewController
+        window.makeKeyAndOrderFront(nil)
+    }
+
+    override func tearDownWithError() throws {
+        dialogProvider = nil
+        factory = nil
+        tab = nil
+        viewController = nil
+        try super.tearDownWithError()
+    }
+
+    func testOnNavigationCompletedIfNoDialogOnlyWebViewVisible() throws {
```

Comment:
Seems in macOS we strictly follow the pattern `testWhen` for test signature names:

```suggestion
    func testWhenNavigationCompletedAndNoDialogThenOnlyWebViewVisible() throws {
```